### ISSUE TITLE
Ability to skip parsing a statement, particularly an eval_gemfile

### ISSUE THREAD

**Issue by lamont-granquist** [dependabot/feedback#720]
We have this our Gemfile in order to let devs add debugging gems (for very complicated reasons we cannot just agree to set a debugging gems and include the globally into a group):

```
# If you want to load debugging tools into the bundle exec sandbox,
# add these additional dependencies into Gemfile.local
eval_gemfile("./Gemfile.local") if File.exist?("./Gemfile.local")
```

I got rid of the old interpolation that was present there, but dependabot still complains now that it can't find Gemfile.local.

What I'd like to be able to do is just tell dependabot to completely ignore that line:

```
# If you want to load debugging tools into the bundle exec sandbox,
# add these additional dependencies into Gemfile.local
eval_gemfile("./Gemfile.local") if File.exist?("./Gemfile.local") # dependabot:ignore
```

We have some other context dependent lines in our Gemfiles that might need to be skipped as well (and it might be useful to be able to skip entire blocks of code with something like `# dependabot:ignore:start` and `# dependabot:ignore:end` or something like that.

**Comment by lamont-granquist** [comment]
Wasn't me, it was actually @tas50 but yeah, we work together.

I think that eval_gemfile is restricted for security reasons.  But that's why I want a magic comment to make your parser just skip this line like it wasn't there.  We don't need you parsing it and don't need to expose you to the security hole.

**Comment by rebelagentm** [comment]
My apologies.  I misunderstood your original post.  We definitely like the idea of adding helpful features.  But the team is pretty swamped right now scaling Dependabot for the GitHub integration, so it may be a while before we can get to doing this.

**Comment by feelepxyz** [comment]
@lamont-granquist @rebelagentm yeah it's currently restricted to prevent untrusted code from executing when we parse the gemfile. We should probably add explicit support for this conditional require pattern instead of an ignore comment. Would love any help implementing this as we're pretty swamped at the moment.

I think we'll want to ignore any eval_gemfiles that are wrapped inside an `if` statement here: https://github.com/dependabot/dependabot-core/blob/master/bundler/lib/dependabot/bundler/file_fetcher/child_gemfile_finder.rb

**Comment by imaizume** [comment]
As reported above, dependabot can't interpret my Gemfile neither.

```ruby
# frozen_string_literal: true
source "https://rubygems.org"

gem 'cocoapods', '~> 1.8.0'
gem 'fastlane', '~> 2.148.0'
gem 'dotenv'

plugins_path = File.join(File.dirname(__FILE__), 'fastlane', 'Pluginfile')
eval_gemfile(plugins_path) if File.exist?(plugins_path)
```

Failure message:

```
Dependabot only supports uninterpolated string arguments to eval_gemfile. Got `plugins_path`
```

> I think we'll want to ignore any eval_gemfiles that are wrapped inside an if statement here:

Like @feelepxyz said, I hope it will come soon.
